### PR TITLE
close #15696

### PR DESCRIPTION
**Pull request by ringabout**
Ref https://github.com/nim-lang/Nim/issues/10815

The first example seems to work while the second one doesn't.

### CODE REVIEW

**Comment by timotheecour** [comment]
> The first example seems to work while the second one doesn't.

then make sure to re-open the issue upon merge as it'll auto-close because of commit msgs

File: tests/vm/tvmmisc.nim [hunk 1]
```diff
@@ -255,6 +255,31 @@ block:
...
+
+  proc `$`(p: Patt): string =
+    discard
+
+  proc P(): Patt =
+    result.add Inst(code: iSet)
+
+  const a = P()
+  doAssert $a == ""
+
```
Comment (timotheecour) [inline tests/vm/tvmmisc.nim:282]:
```suggestion
when defined osx: # xxx bug https://github.com/nim-lang/Nim/issues/10815#issuecomment-476380734
  block:
    type CharSet {.union.} = object 
      cs: set[char]
      vs: array[4, uint64]

    const a = Charset(cs: {'a'..'z'})
    doAssert a.repr.len > 0
```

(can be defered to future PR)

refs https://github.com/nim-lang/Nim/pull/15696#issuecomment-715801619
> looks like it's working on OSX but not on windows or linux i386, amd64
> error: extra brace group at end of initializer

**Review by timotheecour — APPROVED** [review]

**Comment by timotheecour** [comment]
CI failure unrelated (refs https://github.com/nim-lang/Nim/pull/17513)

**Comment by timotheecour** [comment]
@xflywind did the snippet https://github.com/nim-lang/Nim/pull/17518#discussion_r602006428 fail on windows or linux? it works on osx (so should be added as regression test as i showed unless it also works on other platforms)

**Comment by ringabout** [comment]
yeah, it does fail on my win10 machine.

File: tests/vm/tvmmisc.nim (hunk 1 above)
Comment (ringabout) [reply tests/vm/tvmmisc.nim:282]: see https://github.com/nim-lang/Nim/pull/17726

**Pull request merged**